esp/bindings/http/platform/httpservice.cpp

```diff
@@ -401,7 +428,16 @@ int CEspHttpServer::processRequest()
                         m_response->redirect(*m_request.get(),url);
                     }
                     else
+                    {
+                        if (strieq(methodName.str(), "files_") && !checkHttpPathStaysWithinBounds(pathEx))
+                        {
```

Comment:
I am curious where this actually ends up accessing the file?  I would have thought this path just ended up going to onGetFile() which already does the check below.  Is this because some bindings override the onGet() method and don't do their own checks?

Reply:
A path goes to the onGetFile() only if the service name is 'esp' (i.e. GET /esp/files/...). When the service name in a path is not 'esp' (example, .../8010/WsDfu/files_/... ), the EspHttpBinding::onGet() will be called. The EspHttpBinding::onGet() may call the bindings' onGetFile() (for example, CWsDfuSoapBinding::onGetFile()) which doesn't do the check.